modules/programs/hexchat.nix

```
@@ -0,0 +1,248 @@
+              literalExample ''[ "chat.freenode.net" "irc.freenode.net" ]'';
+            default = [ ];
+          };
+          userName = mkOption {
+            type = str;
+            description = "User name";
+          };
+        };
+      };
+    in mkOption {
+      type = nullOr (attrsOf modChannelOption);
+      description = "Configures '~/.config/hexchat/servlist.conf'";
+      example = literalExample ''
+        {
+          freenode = {
+            autojoin = [
+              "#home-manager"
+              "##linux"
+              "#nixos"
+            ];
+            charset = "UTF-8 (Unicode)";
+            commands = [
+              "ECHO Buzz Lightyear sent you a message: 'To Infinity... and Beyond!'"
+            ];
+            loginMethod = sasl;
+            nickname = "my_nickname";
+            nickname2 = "my_secondchoice";
+            options = {
+              acceptInvalidSSLCertificates = false;
+              autoconnect = true;
+              bypassProxy = true;
+              connectToSelectedServerOnly = true;
+              useGlobalUserInformation = false;
+              useSSLForAllServers = false;
+            };
+            password = "my_password";
+            realName = "my_realname";
+            servers = [
+              "chat.freenode.net"
+              "irc.freenode.net"
+            ];
+            userName = "my_username";
+          };
+        }'';
+    };
+    settings = mkOption {
+      description = "Configures '~/.config/hexchat/hexchat.conf'";
+      example = literalExample ''
+        {
+          irc_nick1 = "mynick";
+          irc_username = "bob";
+          irc_realname = "Bart Simpson";
+          text_font = "Monospace 14";
+        };'';
+      type = nullOr (attrsOf str);
+    };
+    enable = mkEnableOption "HexChat - Graphical IRC client";
+    theme = mkOption {
+      description =
+        "Theme package for HexChat. Expects a derivation including decompressed theme files. '.hct' format requires unzip decompression.";
+      type = nullOr package;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.hexchat ];
+    xdg.configFile."hexchat" = mkIf (cfg.theme != null) {
+      source = cfg.theme;
+      recursive = true;
+    };
+    xdg.configFile."hexchat/hexchat.conf" = mkIf (cfg.settings != null) {
+      force = true;
+      text = let hexchatConf = cfg.settings;
+      in concatMapStringsSep "\n" (x: x + " = " + hexchatConf.${x})
+      (attrNames hexchatConf);
+    };
+    xdg.configFile."hexchat/servlist.conf" = mkIf (cfg.channels != null)
+      (if attrNames cfg.channels == [ ] then {
+        text = "";
+      } else {
+        force = true;
+        text = let
+          transformChannel = (channelName:
+            let
+              channel = cfg.channels.${channelName};
+              transformField = (k: v:
+                if v == "" then
+                  ""
+                else ''
+
+                  ${k}=${v}'');
```

```suggestion
                optionalString (v != "") "\n\n${k}=${v}");
```

The formatter might expand the backslash-n unfortunately.

---

Added optionalString converted to '' to satisfy nixfmt.